### DISCUSSION TITLE
The same message, when use rosbag file can show, but use Rosbridge can't show

### DISCUSSION THREAD

**Discussion by doupongzeng** [foxglove/community#61] [Q&A]
I defined some message by myself, and I can `rostopic echo /perception/camera/xxxx` in terminal, this topic has data, but can't show in studio.

When I open a rosbag from file to studio, it's ok to show. But when I use the connection `ws://localhost:9090`, this personal topic can't show, only the other topics(like image) can show.

I also use webviz, has the same problem.
some warning message from use webviz:
```
This can happen e.g. when playing a bag from a different codebase. Message types could not be found for these topics:
/perception/camera/xxxx
/localization/xxxx
....
```

I know this issue because the special types, but use roabag-file studio can show it, so how can I tell the studio to recognize it when I use websocket.

**Comment by jtbandes** [comment]
Which ROS version/distro are you using?

How are you launching the rosbridge server? You will need to make sure the rosbridge server also has access to your custom message types. This might mean you need to launch it in the same ROS workspace as your other code.

**Reply by doupongzeng** [reply]
ROS version: melodic
Ubuntu version: 18.04
I launched the rosbridge server by `roslaunch rosbridge_server rosbridge_websocket.launch`.
I'm sure rosbridge had access my custom message, first I source the setup-file `souce /mypath/setup.bash`, I tested in the same terminal by`rostopic echo /mytopic`, it's can print the currect message, and then I launched the rosbridge.
I'm doubt you said the same ROS workspace, this means I should launch the rosbridge in the ROS workspace path? I try it, the same issue.
What else can I do.
Looking forward to your reply.
